core/lib/zksync_core/src/state_keeper/batch_executor/tests/tester.rs

```diff
@@ -82,45 +88,131 @@ impl Tester {
             db_dir: TempDir::new().unwrap(),
             pool,
             config,
+            tasks: Vec::new(),
         }
     }
 
     pub(super) fn set_config(&mut self, config: TestConfig) {
         self.config = config;
     }
 
-    /// Creates a batch executor instance.
+    /// Creates a batch executor instance with the specified storage type.
     /// This function intentionally uses sensible defaults to not introduce boilerplate.
-    pub(super) async fn create_batch_executor(&self) -> BatchExecutorHandle {
-        // Not really important for the batch executor - it operates over a single batch.
-        let (l1_batch_env, system_env) = self.batch_params(L1BatchNumber(1), 100);
-        self.create_batch_executor_inner(l1_batch_env, system_env)
-            .await
+    pub(super) async fn create_batch_executor(
+        &mut self,
+        storage_type: &StorageType,
```

Comment:
Nit: It's unlikely that this type is changed in the future, so you can implement `Copy` for it and pass it by value.